system-test/bigquery.ts

```diff
@@ -676,6 +677,79 @@ describe('BigQuery', () => {
     });
   });
 
+  describe('BigQuery/Routine', () => {
+    after(async () => {
+      const [routines] = await dataset.getRoutines();
+      return Promise.all(routines.map(routine => routine.delete()));
+    });
+
+    it('should create a routine via insert', () => {
+      return dataset.createRoutine('my_routine', {
+        arguments: [
+          {
+            name: 'x',
+            dataType: {
+              typeKind: 'INT64',
+            },
+          },
+        ],
+        definitionBody: 'x * 3',
+        routineType: 'SCALAR_FUNCTION',
+        returnType: {
+          typeKind: 'INT64',
+        },
+      });
+    });
+
+    it('should create a routine via DDL', () => {
+      const routineId = `${bigquery.projectId}.${dataset.id}.my_ddl_routine`;
+
+      return bigquery.query(`
+        CREATE FUNCTION \`${routineId}\`(
+          arr ARRAY<STRUCT<name STRING, val INT64>>
+        ) AS (
+          (SELECT SUM(IF(elem.name = "foo",elem.val,null)) FROM UNNEST(arr) AS elem)
+        )
+      `);
+    });
+
+    it('should list all the routines', async () => {
```

Comment:
This shouldn't depend on the previous tests. I think the ways I have handled this in the past is:

- Create an object in the before() hook, using the first `it` to confirm the name/other metadata was properly set. Then, in a list test, just check that >= 1 exists.

- Create an object in the list test, then confirm the list returns >= 1.